dashboard/modules/serve/serve_head.py

```diff
@@ -0,0 +1,70 @@
+from aiohttp.web import Request, Response
+import json
+import logging
+
+import ray.dashboard.utils as dashboard_utils
+import ray.dashboard.optional_utils as optional_utils
+
+from ray import serve
+from ray.serve.api import deploy_group
+from ray.dashboard.modules.serve.schema import (
+    ServeInstanceSchema,
+    serve_instance_to_schema,
+    schema_to_serve_instance,
+)
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+routes = optional_utils.ClassMethodRouteTable
+
+
+class ServeHead(dashboard_utils.DashboardHeadModule):
+    def __init__(self, dashboard_head):
+        super().__init__(dashboard_head)
+
+    @routes.get("/api/serve/deployments/")
+    @optional_utils.init_ray_and_catch_exceptions(connect_to_serve=True)
+    async def get_all_deployments(self, req: Request) -> Response:
+        deployments = list(serve.list_deployments().values())
+        serve_instance_schema = serve_instance_to_schema(deployments)
+        return Response(
+            text=json.dumps(serve_instance_schema.json()),
+            content_type="application/json",
+        )
+
+    @routes.delete("/api/serve/deployments/")
+    @optional_utils.init_ray_and_catch_exceptions(connect_to_serve=True)
+    async def delete_all_deployments(self, req: Request) -> Response:
+        serve.shutdown()
```

Comment:
is this actually what we want to do or should we instead just delete the deployments themselves? this behavior seems ok to me given that we aren't accepting namespaces or anything

Reply:
I think `serve.shutdown()` makes sense because then a DELETE request deletes the entire serve instance, not just the deployments. I've renamed the function to `delete_serve_instance` to reflect that.

Reply:
ok works for me